solc/CommandLineInterface.cpp

```diff
@@ -1094,6 +1167,55 @@ void CommandLineInterface::assembleYul(yul::YulStack::Language _language, yul::Y
 			else
 				serr() << "No text representation found." << endl;
 		}
+
+		if (m_options.compiler.outputs.asmJson)
+		{
+			shared_ptr<evmasm::Assembly> assembly{stack.assembleEVMWithDeployed().first};
+			if (assembly)
+			{
+				function<map<string, unsigned>(yul::Object const&)> collectSourceIndices =
+					[&](yul::Object const& _object) -> map<string, unsigned> {
+					map<string, unsigned> sourceIndices;
+					if (_object.debugData && _object.debugData->sourceNames.has_value())
+						for (auto const& iter: *_object.debugData->sourceNames)
+							sourceIndices[*iter.second] = iter.first;
+					for (auto const& sub: _object.subObjects)
+					{
+						auto subObject = dynamic_cast<yul::Object const*>(sub.get());
+						if (subObject)
+							for (auto const& [name, index]: collectSourceIndices(*subObject))
+								sourceIndices[name] = index;
+					}
+					return sourceIndices;
+				};
+				if (stack.parserResult() && stack.parserResult()->debugData)
+				{
+					map<string, unsigned> sourceIndices = collectSourceIndices(*stack.parserResult());
+					// if sourceIndices are empty here, there were no source locations annotated in the yul source.
+					// in this case, we just add the filename of the yul file itself here.
+					if (sourceIndices.empty())
+						sourceIndices[src.first] = 0;
+					size_t max_index = 0;
+					for (auto const& [name, index]: sourceIndices)
+						if (max_index < index)
+							max_index = index;
```

Comment:
```suggestion
						max_index = max(index, max_index);
```
`std::max`, so might need to include `<algorithm>`